dart/lib/src/client.dart

```diff
@@ -19,38 +18,19 @@ abstract class SentryClient {
   /// `dart:html` is available, otherwise it will throw an unsupported error.
   factory SentryClient(SentryOptions options) => createSentryClient(options);
 
-  SentryClient.base(
-    this.options, {
-    String platform,
-    this.origin,
-    Sdk sdk,
-  })  : _dsn = Dsn.parse(options.dsn),
-        _platform = platform ?? sdkPlatform,
-        sdk = sdk ?? Sdk(name: sdkName, version: sdkVersion);
-
-  final Dsn _dsn;
-
-  @protected
-  SentryOptions options;
-
-  /// The DSN URI.
-  @visibleForTesting
-  Uri get dsnUri => _dsn.uri;
-
-  /// The Sentry.io public key for the project.
-  @visibleForTesting
-  // ignore: invalid_use_of_visible_for_testing_member
-  String get publicKey => _dsn.publicKey;
+  SentryClient.base(this.options, {String origin}) {
+    _random = options.sampleRate == null ? null : Random();
+    if (options.transport is NoOpTransport) {
+      options.transport = Transport(
+        options: options,
+        sdkIdentifier: '${sdkName}/${sdkVersion}',
```

Comment:
could this be replaced by `options.sdkVersion.identifier`? it looks duplicated now

Reply:
✅ my bad, I thought `sdkVersion` was just a String, I renamed it `sdk`, and define it in the ioClient & BrowserClient if null